samples/sensor/bme280/src/main.c

```diff
@@ -37,23 +49,65 @@ static const struct device *get_bme280_device(void)
 
 int main(void)
 {
-	const struct device *dev = get_bme280_device();
+	const struct device *dev = check_bme280_device();
 
 	if (dev == NULL) {
 		return 0;
 	}
 
 	while (1) {
-		struct sensor_value temp, press, humidity;
+		uint8_t buf[128];
+
+		int rc = sensor_read(&iodev, &ctx, buf, 128);
+
+		if (rc != 0) {
+			printk("%s: sensor_read() failed: %d\n", dev->name, rc);
+			return rc;
+		}
+
+		const struct sensor_decoder_api *decoder;
+
+		rc = sensor_get_decoder(dev, &decoder);
+
+		if (rc != 0) {
+			printk("%s: sensor_get_decode() failed: %d\n", dev->name, rc);
+			return rc;
+		}
+
+		uint32_t temp_fit = 0;
+		uint8_t temp_buf[64];
+
+		decoder->decode(buf,
+			(struct sensor_chan_spec) {SENSOR_CHAN_AMBIENT_TEMP, 0},
+			&temp_fit, 1, temp_buf);
+
+		struct sensor_q31_data *temp_data =
+				(struct sensor_q31_data *) temp_buf;
+
+		uint32_t press_fit = 0;
+		uint8_t press_buf[64];
+
+		decoder->decode(buf,
+				(struct sensor_chan_spec) {SENSOR_CHAN_PRESS, 0},
+				&press_fit, 1, press_buf);
+
+		struct sensor_q31_data *press_data =
+				(struct sensor_q31_data *) press_buf;
+
+		uint32_t hum_fit = 0;
+		uint8_t hum_buf[64];
```

Comment:
Is there a scenario where this won't be `struct sensor_q31_data`? If not, I'd suggest directly using that data-type.

Same suggestion applies for `press_buf` and `temp_buf`.

Reply:
Good eye, the values to decode into should indeed by q31's

Reply:
No, and I've changed to use the data type directly.